pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -0,0 +1,779 @@
+        CompletableFuture<String> getOwnerRequest = deferGetOwnerRequest(serviceUnit);
+        pubAsync(serviceUnit, new ServiceUnitStateData(Assigned, broker))
+                .whenComplete((__, ex) -> {
+                    if (ex != null) {
+                        getOwnerRequests.remove(serviceUnit);
+                        if (!getOwnerRequest.isCompletedExceptionally()) {
+                            getOwnerRequest.completeExceptionally(ex);
+                        }
+                    }
+                });
+
+        return getOwnerRequest;
+    }
+
+    public CompletableFuture<Void> publishUnloadEventAsync(Unload unload) {
+        String serviceUnit = unload.serviceUnit();
+        if (isTransferCommand(unload)) {
+            ServiceUnitStateData next = new ServiceUnitStateData(Assigned,
+                    unload.destBroker().get(), unload.sourceBroker());
+            return pubAsync(serviceUnit, next).thenAccept(__ -> {
+            });
+        }
+        return tombstoneAsync(serviceUnit).thenAccept(__ -> {
+        });
+    }
+
+    public CompletableFuture<Void> publishSplitEventAsync(Split split) {
+        String serviceUnit = split.serviceUnit();
+        ServiceUnitStateData data = tableview.get(serviceUnit);
+        ServiceUnitStateData next = new ServiceUnitStateData(Splitting, data.broker());
+        return pubAsync(serviceUnit, next).thenAccept(__ -> {
+        });
+    }
+
+    private void handle(String serviceUnit, ServiceUnitStateData data) {
+        if (log.isDebugEnabled()) {
+            log.info("{} received a handle request for serviceUnit:{}, data:{}",
+                    lookupServiceAddress, serviceUnit, data);
+        }
+
+        if (data == null) {
+            handleTombstoneEvent(serviceUnit);
+            return;
+        }
+
+        // TODO : Add state validation
+        switch (data.state()) {
+            case Owned -> handleOwnEvent(serviceUnit, data);
+            case Assigned -> handleAssignEvent(serviceUnit, data);
+            case Released -> handleReleaseEvent(serviceUnit, data);
+            case Splitting -> handleSplitEvent(serviceUnit, data);
+            default -> throw new IllegalStateException("Failed to handle channel data:" + data);
+        }
+    }
+
+    private static boolean isTransferCommand(ServiceUnitStateData data) {
+        if (data == null) {
+            return false;
+        }
+        return StringUtils.isNotEmpty(data.sourceBroker());
+    }
+
+    private static boolean isTransferCommand(Unload data) {
+        return data.destBroker().isPresent();
+    }
+
+    private static String getLogEventTag(ServiceUnitStateData data) {
+        return data == null ? "Free" :
+                isTransferCommand(data) ? "Transfer:" + data.state() : data.state().toString();
+    }
+
+    private void log(Throwable e, String serviceUnit, ServiceUnitStateData data, ServiceUnitStateData next) {
```

Comment:
nit: static ?

Reply:
No. It can't. `lookupServiceAddress` is a member var.